src/instance/minecraft/chat-manager.ts

```diff
@@ -123,4 +129,35 @@ export default class ChatManager extends EventHandler<MinecraftInstance> {
       message
     })
   }
+
+  proceedFiltering(author: string, content: string): string {
+    let filteredMessage: string
+    try {
+      filteredMessage = this.clientInstance.app.profanityFilter.clean(content)
+    } catch {
+      /*
+        profanity package has bug.
+        will throw error if given one special character.
+        example: clean("?")
+        message is clear if thrown
+      */
+      filteredMessage = content
+    }
+
+    if (content !== filteredMessage) {
+      this.clientInstance.app.emit('event', {
+        localEvent: true,
+        instanceType: InstanceType.DISCORD,
+        username: author,
+        message: `${escapeDiscord(author)}:\n\n**Profanity warning, this message has been edited:**\n${escapeDiscord(filteredMessage)}`,
+        instanceName: InstanceType.MAIN,
+        eventType: EventType.AUTOMATED,
```

Comment:
`EventType.AUTOMATED` is documented to be only used by plugins. Its sole purpose is to allow plugins to send events that aren't already categorized by the application. For a core feature, a new event or event type must be created instead.

Reply:
`event` event is an old event that was meant to for in-game chat events. The event is used to send identified chat messages such as when a message says that a player joined the guild, or when a message was blocked for repetitive, etc.
The event is just a 1:1 raw chat message to an in-game parsed event. Using it to send an auxiliary events that aren't 1:1 with raw chat isn't how it is used.

The event name `event` is kept for backward compatibility. In true sense, the event should be called `IngameChatEvent`.

As to where discord parses THIS particular event: https://github.com/aidn3/hypixel-guild-discord-bridge/blob/d1c82cef5273d600657a8b50fd301d2ddc4e5bb2/src/instance/discord/discord-instance.ts#L59-L61 


What you need to do instead, is create a new event and use it instead of this one. Then handle the event from discord side the same way other events are handled.

Your new event should inherit the base `InformEvent` of course: https://github.com/aidn3/hypixel-guild-discord-bridge/blob/d1c82cef5273d600657a8b50fd301d2ddc4e5bb2/src/common/application-event.ts#L117-L130